solenoid_app/solenoid_math/solver.py

```diff
@@ -107,55 +107,187 @@ def solenoid_solve(volts, length, r0, rA, gauge, location, force):
+        result = solve(origin_eq, RA)
+
+    elif location is None:
+        # TODO: handle with error (here or at frontend)
+        origin_eq = origin_eq.subs(v, volts)
+        origin_eq = origin_eq.subs(R0, r0)
+        origin_eq = origin_eq.subs(RA, ra)
+        origin_eq = origin_eq.subs(l, length)
+        origin_eq = origin_eq.subs(f, force)
+        # result = solve(origin_eq, x)
+        result = -1
+
     elif force is None:  
-        origin_eq = origin_eq.subs(v,volts)
-        origin_eq = origin_eq.subs(R0,r0)
-        origin_eq = origin_eq.subs(RA,rA)
-        origin_eq = origin_eq.subs(x,location)
-        origin_eq = origin_eq.subs(l,length)
-        result = solve(origin_eq,f)
+        origin_eq = origin_eq.subs(v, volts)
+        origin_eq = origin_eq.subs(R0, r0)
+        origin_eq = origin_eq.subs(RA, ra)
+        origin_eq = origin_eq.subs(x, location)
+        origin_eq = origin_eq.subs(l, length)
+        result = solve(origin_eq, f)
 
     # Return the correctly signed value due to solving squares in function
     if len(result) == 2:
-        return result[1]
+        return float(result[1])
     else:
-        return result[0]
+        return float(result[0])
+
+
+"""
+Solves directly for a single missing variable within the solenoid force equation.
+
+Performs much faster than the sympy equivalents in solenoid_solve
+
+The variable being solved for is inputted as a None value. All other
+arguments are then required and cannot be None
+
+Parameters:
+    volts (float | None): Voltage applied to the solenoid - Volts
+    length (float | None): Overall length of the solenoid coil - Millimeters
+    r0 (float | None): Inner radius of the solenoid coil - Millimeters
+    ra (float | None): Outer radius of the solenoid coil - Millimeters
+    gauge (string): A value between "0000" -> "40"
+    location (float | None): Location (Stroke) of the solenoid core within the coil - Millimeters
+    force (float | None): The force produced by the solenoid - Newtons
+
+Returns:
+    result (float): The solved value of specified variable
+"""
+def solenoid_performance(volts, length, r0, ra, gauge, location, force):
+    result = None
+    a = np.log(PERM_RELATIVE)
+
+    if volts is None:
+
+        result = (2 * np.sqrt(2 * np.pi) * np.sqrt(force) * (AWG_DATA[gauge]["resistance"] / 1000) * (length / 1000) * (
+                    ra / 1000) * np.e ** (((location / 1000) * a) / (2 * (length / 1000)))) / (
+                             (r0 / 1000) * np.sqrt(PERM_RELATIVE) * np.sqrt(PERM_FREE) * np.sqrt(a))
+
+    # TODO: Find EQ or handle location != 0 case with error (probably frontend)
+    elif length is None:
+
+        if location == 0:
+            result = 1000 * (np.sqrt(a) * (r0 / 1000) * np.sqrt(PERM_FREE) * np.sqrt(PERM_RELATIVE) * volts) / (
+                    2 * np.sqrt(2 * np.pi) * np.sqrt(force) * (AWG_DATA[gauge]["resistance"] / 1000) * (ra / 1000))
+        else:
+            result = -1
```

Comment:
We might not need this check if we can filter out the input on the front end.  Just a small note.